Description: Adjusts the compiler flags to enable the hardening and link to the system libraries
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/buildSrc/linux.gradle
+++ b/buildSrc/linux.gradle
@@ -53,7 +53,9 @@
         commonFlags, "-I$JDK_HOME/include", "-I$JDK_HOME/include/linux", "-c",
         IS_DEBUG_NATIVE ? ["-ggdb", "-DVERBOSE"] : ["-O2", "-DNDEBUG"]].flatten()
 //ccFlags.addAll(["-Wnon-virtual-dtor", "-Woverloaded-virtual", "-std=c++0x"])
+ccFlags.addAll(["-fstack-protector", "--param=ssp-buffer-size=4", "-Wformat", "-Werror=format-security", "-D_FORTIFY_SOURCE=2"])
 def linkFlags = ["-shared", commonFlags].flatten()
+linkFlags.addAll(["-Wl,-z,relro"])
 
 // Create $buildDir/linux_tools.properties file and load props from it
 setupTools("linux_tools",
@@ -182,7 +184,7 @@
 LINUX.iio.compiler = compiler
 LINUX.iio.ccFlags = [ccFlags].flatten()
 LINUX.iio.linker = linker
-LINUX.iio.linkFlags = [linkFlags].flatten()
+LINUX.iio.linkFlags = [linkFlags, "-ljpeg"].flatten()
 LINUX.iio.lib = "javafx_iio"
 
 LINUX.prismES2 = [:]
